CASMINST-3902: Add and update procedures related to adding river cabinets to a system (#2344)
Cray-HPE/docs-csm: operations/node_management/Update_the_HPE_Node_BIOS_Time.md

# Update the HPE Node BIOS Time

Check and set the time for HPE nodes.

> If the console log indicates the time between the rest of the system and the node is off by several hours, it prevents the spire-agent from getting a valid certificate, causing the node boot to drop into the dracut emergency shell.

Author: https://github.com/Cray-HPE/docs-csm/commit/ab3c45572fa5b86e6e61bbd196c6734919b6dae8#diff-77a8a9c40e37b9227e26cafc5bf523b5841d0e705bbd5c5c143b8ed99d87e17aR5

## Procedure

1. Log in to a second terminal session in order to watch the node's console.

>     ***Please open this link in a new tab or page*** [Log in to a Node Using ConMan](../conman/Log_in_to_a_Node_Using_ConMan.md)

Author: https://github.com/Cray-HPE/docs-csm/commit/ab3c45572fa5b86e6e61bbd196c6734919b6dae8#diff-77a8a9c40e37b9227e26cafc5bf523b5841d0e705bbd5c5c143b8ed99d87e17aR11

    The first terminal session will be needed to run the commands to boot the node into the BIOS menu.

> 1. (`ncn#`) In another terminal session set the following environment variables:

Reviewer: ```suggestion
1. (`ncn#`) In another terminal session, set environment variables.
```

Author: https://github.com/Cray-HPE/docs-csm/commit/ab3c45572fa5b86e6e61bbd196c6734919b6dae8#diff-77a8a9c40e37b9227e26cafc5bf523b5841d0e705bbd5c5c143b8ed99d87e17aR15

    1. The BMC hostname.

        ```bash
        BMC=x3000c0s3b0
        ```

    1. Export the `root` user password of the BMC.

        > NOTE: `read -s` is used to prevent the password from echoing to the screen or
        > being saved in the shell history.

        ```bash
        read -r -s -p "${BMC} root password: " IPMI_PASSWORD
        export IPMI_PASSWORD
        ```

1. (`ncn#`) Boot the node into its BIOS via the serial console.

>     When `ipmitool` is used to boot a HPE node into its BIOS menu it will not be available on on the nodes serial console, due to the node booting into a graphical BIOS menu. To access the serial version of the BIOS setup. Perform the `ipmitool` steps

Reviewer: ```suggestion
    When `ipmitool` is used to boot a HPE node into its BIOS menu, it will not be available on the node's serial console, because of the node booting into a graphical BIOS menu. In order to access the serial version of the BIOS setup, perform the `ipmitool` steps
```

Author: https://github.com/Cray-HPE/docs-csm/commit/ab3c45572fa5b86e6e61bbd196c6734919b6dae8#diff-77a8a9c40e37b9227e26cafc5bf523b5841d0e705bbd5c5c143b8ed99d87e17aR35

>     above to boot the node. Then in conman press `ESC+9` key combination when you see the following messages in the console. This will open a menu you use to enter the BIOS via serial console.

Reviewer: ```suggestion
    above to boot the node. Then, on the serial console, enter the BIOS by pressing the `ESC+9` key combination when the following messages are shown on the console:
```

Author: https://github.com/Cray-HPE/docs-csm/commit/ab3c45572fa5b86e6e61bbd196c6734919b6dae8#diff-77a8a9c40e37b9227e26cafc5bf523b5841d0e705bbd5c5c143b8ed99d87e17aR36

    ```text
    For access via BIOS Serial Console:
    Press 'ESC+9' for System Utilities
    Press 'ESC+0' for Intelligent Provisioning
    Press 'ESC+!' for One-Time Boot Menu
    Press 'ESC+@' for Network Boot
    ```

>     ```bash

Reviewer: Earlier the text refers to the "above" ipmitool commands. Should these be above that?

Author: I've changed the wording to the following: https://github.com/Cray-HPE/docs-csm/pull/2344/commits/d95c1e8bf36de5c16447a201cb2ecf3e6f4b5113

    ipmitool -I lanplus -U root -E -H $BMC chassis power off
    sleep 10
    ipmitool -I lanplus -U root -E -H $BMC chassis bootdev bios
    ipmitool -I lanplus -U root -E -H $BMC chassis power on
    ```

>     > (`linux#`) Alternatively, for HPE NCNs you can log in to the BMC's web interface and access the HTML5 console for the node, in order to interact with the graphical BIOS.

Reviewer: ```suggestion
    > (`linux#`) Alternatively, for HPE NCNs, it is possible to access the HTML5 console for the node by using the BMC's web interface. This works with the graphical BIOS.
```

Author: Applied here: https://github.com/Cray-HPE/docs-csm/pull/2344/commits/ab3c45572fa5b86e6e61bbd196c6734919b6dae8#diff-77a8a9c40e37b9227e26cafc5bf523b5841d0e705bbd5c5c143b8ed99d87e17aR53

    > From the administrator's own machine, create an SSH tunnel (`-L` creates the tunnel; `-N` prevents a shell and stubs the connection):
    >
    > ```bash
    > bmc=x3000c0s3b0 # Change this to be each node in turn.
    > ssh -L 9443:$bmc:443 -N root@eniac-ncn-m001
    > ```
    >
    > Opening a web browser to `https://localhost:9443` will give access to the BMC's web interface.

> 1. When the node boots, you will be able to use the conman session to see the BIOS menu to check and set the time to current UTC time. The process varies depending on the vendor of the NCN.

Reviewer: ```suggestion
1. When the node boots, use the console session to check and set the time in the BIOS to the current UTC time.
```

Author: Applied here: https://github.com/Cray-HPE/docs-csm/pull/2344/commits/ab3c45572fa5b86e6e61bbd196c6734919b6dae8#diff-77a8a9c40e37b9227e26cafc5bf523b5841d0e705bbd5c5c143b8ed99d87e17aR63

>     On HPE NCNs the date configuration menu can be found at the following path: `System Configuration -> BIOS/Platform Configuration (RBSU) -> Date and Time`

Reviewer: ```suggestion
    On HPE NCNs, the date configuration menu is at the following path: `System Configuration -> BIOS/Platform Configuration (RBSU) -> Date and Time`
```

Author: Applied here: https://github.com/Cray-HPE/docs-csm/pull/2344/commits/ab3c45572fa5b86e6e61bbd196c6734919b6dae8#diff-77a8a9c40e37b9227e26cafc5bf523b5841d0e705bbd5c5c143b8ed99d87e17aR65

> 1. (`ncn#`) After you have verified the correct time, power off the NCN.

Reviewer: ```suggestion
1. (`ncn#`) After verifying the correct time, power off the NCN.
```

Author: Applied here: https://github.com/Cray-HPE/docs-csm/pull/2344/commits/ab3c45572fa5b86e6e61bbd196c6734919b6dae8#diff-77a8a9c40e37b9227e26cafc5bf523b5841d0e705bbd5c5c143b8ed99d87e17aR67
